### PR TITLE
fix: allow multiple nested groups

### PR DESCRIPTION
**Pull request by rephus**
Closes:  https://github.com/lightdash/lightdash/issues/11316

### Description:

image

 
image

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: examples/full-jaffle-shop-demo/dbt/models/events.yml [hunk 1]
```diff
@@ -68,7 +72,7 @@ models:
         meta:
           dimension:
             type: number
-            groups: ['events']
+            groups: ['events', 'nested_group', 'another_nested_group']
```
Comment (rephus) [inline examples/full-jaffle-shop-demo/dbt/models/events.yml:75]: I'm not sure if we want to keep this , or keep it as it was.

**Review by ZeRego — APPROVED** [review]

File: packages/frontend/src/components/Explorer/ExploreTree/TableTree/Tree/TreeProvider.tsx [hunk 2]
```diff
@@ -37,7 +37,7 @@ const isBaseDimensionWithIntervalDefined = (item: Item): boolean => {
     }
 };
 
-const MAX_GROUP_DEPTH = 2;
+const MAX_GROUP_DEPTH = 3;
```
Comment (ZeRego) [inline packages/frontend/src/components/Explorer/ExploreTree/TableTree/Tree/TreeProvider.tsx:40]: do we need to update docs?

Comment (rephus) [reply packages/frontend/src/components/Explorer/ExploreTree/TableTree/Tree/TreeProvider.tsx:40]: yeah, will do

**Comment by rephus** [comment]
Docs: https://deploy-preview-12233--peaceful-bassi-cbf284.netlify.app/references/dimensions#groups

**Pull request merged**